File: VB/RangeDate/MyGridControl/MyGridView.vb
Imports System
Imports System.Linq
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.Utils.Serializing
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.Data.Filtering
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls


Namespace DateRange
    <System.ComponentModel.DesignerCategory("")> _
    Public Class MyGridView
        Inherits GridView

        Public Sub New()
            Me.New(Nothing)
        End Sub
        Public Sub New(ByVal grid As GridControl)
            MyBase.New(grid)

        End Sub
        Protected Overrides Function CreateColumnCollection() As GridColumnCollection
            Return New MyGridColumnCollection(Me)
        End Function
        <Browsable(False), XtraSerializableProperty(XtraSerializationVisibility.Collection, True, True, True, 0, XtraSerializationFlags.DefaultValue), XtraSerializablePropertyId(2), DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Shadows ReadOnly Property Columns() As MyGridColumnCollection
            Get
                Return TryCast(MyBase.Columns, MyGridColumnCollection)
            End Get
        End Property

        Protected Overrides Function CreateDateFilterPopup(ByVal column As GridColumn, ByVal ownerControl As System.Windows.Forms.Control, ByVal creator As Object) As DateFilterPopup
            Return New MyDateFilterPopup(Me, column, ownerControl, creator)
        End Function

        Private Enum some
            Greater
            DateRange
        End Enum

        Protected Overrides Sub RaiseFilterPopupDate(ByVal filterPopup As DateFilterPopup, ByVal list As List(Of FilterDateElement))
            Dim filter As CriteriaOperator = New BinaryOperator(filterPopup.Column.FieldName, Date.Today, BinaryOperatorType.Greater)

            list.Add(New FilterDateElement(Localizer.Active.GetLocalizedString(StringId.FilterClauseGreater), "", filter))
            filter = New BinaryOperator(filterPopup.Column.FieldName, Date.Today, BinaryOperatorType.Less)
            list.Add(New FilterDateElement(Localizer.Active.GetLocalizedString(StringId.FilterClauseLess), "", filter))
            filter = New BetweenOperator(filterPopup.Column.FieldName, Date.Today, Date.Today)
            list.Add(New FilterDateElement(Localizer.Active.GetLocalizedString(StringId.FilterClauseBetween), "", filter))
            MyBase.RaiseFilterPopupDate(filterPopup, list)
        End Sub
    End Class
End Namespace

File: CS/RangeDate/MyGridControl/MyGridView.cs
using System;
using System.Linq;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.Utils.Serializing;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.Data.Filtering;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;


namespace DateRange {
    [System.ComponentModel.DesignerCategory("")]
    public class MyGridView : GridView {
        public MyGridView() : this(null) { }
        public MyGridView(GridControl grid)
            : base(grid) {

        }
        protected override GridColumnCollection CreateColumnCollection() {
            return new MyGridColumnCollection(this);
        }
        [Browsable(false)]
        [XtraSerializableProperty(XtraSerializationVisibility.Collection, true, true, true, 0, XtraSerializationFlags.DefaultValue)]
        [XtraSerializablePropertyId(2)]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new MyGridColumnCollection Columns {
            get { return base.Columns as MyGridColumnCollection; }
        }

        protected override DateFilterPopup CreateDateFilterPopup(GridColumn column, System.Windows.Forms.Control ownerControl, object creator) {
            return new MyDateFilterPopup(this, column, ownerControl, creator);
        }

        enum some { Greater, DateRange }

        protected override void RaiseFilterPopupDate(DateFilterPopup filterPopup, List<FilterDateElement> list) {
            CriteriaOperator filter = new BinaryOperator(filterPopup.Column.FieldName, DateTime.Today, BinaryOperatorType.Greater);

            list.Add(new FilterDateElement(Localizer.Active.GetLocalizedString(StringId.FilterClauseGreater)
                , "", filter));
            filter = new BinaryOperator(filterPopup.Column.FieldName, DateTime.Today, BinaryOperatorType.Less);
            list.Add(new FilterDateElement(Localizer.Active.GetLocalizedString(StringId.FilterClauseLess)
                , "", filter));
            filter = new BetweenOperator(filterPopup.Column.FieldName, DateTime.Today, DateTime.Today);
            list.Add(new FilterDateElement(Localizer.Active.GetLocalizedString(StringId.FilterClauseBetween), "", filter));
            base.RaiseFilterPopupDate(filterPopup, list);
        }
    }
}